Cron/Clean.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Lfi\MonologDatabase\Cron;
+
+use Magento\Framework\App\Config\ScopeConfigInterface;
+use Lfi\MonologDatabase\Api\Data\RecordInterface;
+use Lfi\MonologDatabase\Model\ResourceModel\Record;
+
+class Clean
+{
+    protected $config;
+    protected $resource;
+
+
+    public function __construct(ScopeConfigInterface $config, Record $resource)
+    {
+        $this->config = $config;
+        $this->resource = $resource;
+    }
+
+    /**
+     * Get the record lifetime threshold based on the number of days.
+     * @param int $days
+     * @return string
+     */
+    protected function getThreshold(int $days): string
+    {
+        $dt = new \DateTime();
+        $offset = new \DateInterval("P{$days}D");
```

Comment:
`DateInterval`'s [constructor](https://www.php.net/manual/en/dateinterval.construct.php) is picky when it comes to input. We ought to place a validity check on the `lifetime` config field either on save or load to ensure it's actually a positive int.

Reply:
Good point, I forced `NULL` to `int` so it would be `0` but didn’t consider that someone might enter a negative numeric value which could trigger an error. I will also set a better default instead. 👍